src/preimages.jl

```diff
@@ -166,39 +167,63 @@ end
 """
 function preimages(y, D::Dynamic, ylabel = 1:length(y); ϵ, max_iter)
     results = @showprogress 1 "Computing preimages..." [preimages(y, b, ylabel; ϵ, max_iter) for b in branches(D)]
-    x = vcat((result[1] for result in results)...)
-    xlabel = vcat((result[2] for result in results)...)
+    x = reduce(vcat, result[1] for result in results)
+    xlabel = reduce(vcat, result[2] for result in results)
     return x, xlabel
 end
 
 """
-    preimages_and_derivatives(y, br::MonotonicBranch, ylabel = 1:length(y), ϵ = 0.0)
+    preimages_and_branches(y, D::Dynamic, ylabel = 1:length(y); ϵ, max_iter)
+
+Works like preimages(y, D) but the second return argument is a vector of pairs (ylabel, branchnumber) that specifies
+which branch of D is the source of each interval in the preimages.
+"""
+# TODO: there is probably a very clever general construction that can reduce `preimages`, `preimages_and_branches`, 
```

Comment:
I don't think doing all the Newton step directly on the Taylor1 type is a good solution, unfortunately. In this way you would have to compute many more derivatives: in each iterate $x_i$ of the Newton step you'd have to compute the corresponding $f'(x_i)$ rather than only $f(x_i)$. But this is not necessary, you only need to compute the derivative in the final point at which Newton converges.

What I had in mind instead is writing `preimages` so that it propagates generic "additional information" in the form of a function `f(branch, index_k, label)` across compositions. In the case of `preimages_and_derivatives`, this information is the derivative. But this might be overkill, since we only need 2-3 different cases for this additional information.

Another possible solution (which would be handy also for the C2 basis in future) is having a method `preimage(y::Taylor1, br, ...)` which computes `preimage` as usual and then afterwards computes the additional derivatives that are requested. Or a "differentiation rule for `preimage`", for a different differentiation backend. But this might be overkill, too; for now what we have works so I'd suggest to leave it for the future.